homepage/models.py

```diff
@@ -125,10 +184,54 @@ def get_elective_courses(courses):
         # gets all courses that are electives
         return courses.filter(mandatory=False)
 
-    # --- returns if course has Prerequisites
-    def has_preqs(self):
-        return Prerequisites.does_course_have_prerequisites(self)
+    @staticmethod
+    def get_courses_with_preqs(courses):
+        # gets all courses that have prerequisites
+        have_preqs = []
+        for course in courses:
+            if course.has_preqs():
+                have_preqs.append(course.course_id)
+        return courses.filter(course_id__in=have_preqs)
+
+    @staticmethod
+    def get_courses_without_preqs(courses):
+        # gets all courses that have prerequisites
+        no_preqs = []
+        for course in courses:
+            if not course.has_preqs():
+                no_preqs.append(course.course_id)
+        return courses.filter(course_id__in=no_preqs)
+
+    @staticmethod
+    def get_courses_with_ratings(courses, num):

```

Comment:
maybe change the name of the input parameter 'num' to num_of_ratings

Reply:
changed to 'at_least'